commit 471e9e3e02f144a3c5f2aaf6a48be541a92ff44c
Author:     Chester Chen <512707+chesterxgchen@users.noreply.github.com>
AuthorDate: Mon Sep 19 19:40:34 2022 -0700
Commit:     GitHub <noreply@github.com>
CommitDate: Mon Sep 19 19:40:34 2022 -0700

    Fed stats privacy2 [skip ci] (#897)
    
    * Fed STATS FILTER AND PRIVACY
    Fed_statics_privacy polocies
    update privacy filters
    rename directory
    refactoring and add privacy tests
    optimize imports
    update README.md
    change filter order
    
    * 1) rename the metrics_privacy_filter.py to metrics_privacy_cleanser.py
    2)  renmae MetricsPrivacyFilter to MetricsPrivacyCleanser
    rename xxx_check to xxx_checker
    
    * Based on the PR feedbacks, made the following chnages
    1) move the MetricsPrivacyCleanser to app_common/statistics instead keep it in app_common/filters/statstics
    2) since all these cleansers are not longer in above package removed the filter/stastics module and directly move the statistics_privacy_filter.py  to app_common/filters
    3) update the config_fed_client.json to use task_result_filters for privacy, this way the fed stats can run in simulator with job configuration directly
    4) update the read me to include both privacy.json and job privacy filter configuration
    
    * update README.md
---
 nvflare/private/fed/server/job_runner.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nvflare/private/fed/server/job_runner.py b/nvflare/private/fed/server/job_runner.py
index 8d1ed9a6..681b1568 100644
--- a/nvflare/private/fed/server/job_runner.py
+++ b/nvflare/private/fed/server/job_runner.py
@@ -186,7 +186,7 @@ class JobRunner(FLComponent):
                             deploy_detail.append(f"failed to deploy to required client {c}")
 
         if abort_job:
-            raise RuntimeError("deploy failure")
+            raise RuntimeError("deploy failure", deploy_detail)
 
         self.fire_event(EventType.JOB_DEPLOYED, fl_ctx)
         return run_number
